src/components/ContactsApp.jsx

```diff
@@ -2,16 +2,47 @@ import React from "react";
 import ContactsList from "./ContactsList/ContactsList";
 import { withContacts } from "./ContactsList";
 import ContactsHeader from "./ContactsList/ContactsHeader";
+import ContactCard from "./ContactCard/ContactCard";
+import Modal, { ModalContent } from "cozy-ui/react/Modal";
 
 const ConnectedContactsList = withContacts(ContactsList);
 
-const ContactsApp = () => (
-  <main className="app-content">
-    <ContactsHeader />
-    <div role="contentinfo">
-      <ConnectedContactsList />
-    </div>
-  </main>
-);
+class ContactsApp extends React.Component {
+  state = {
+    displayedContact: null
+  };
 
-export default ContactsApp;
+  showContact = contact => {
+    this.setState({
+      displayedContact: contact
+    });
+  };
+
+  hideContact = () => {
+    this.setState({
+      displayedContact: null
+    });
+  };
+
+  render() {
+    const { displayedContact } = this.state;
+
+    return (
+      <main className="app-content">
+        <ContactsHeader />
+        <div role="contentinfo">
+          <ConnectedContactsList onClickContact={this.showContact} />
+        </div>
+        {displayedContact && (
+          <Modal into="body" dismissAction={this.hideContact}>
+            <ModalContent>
+              <ContactCard contact={displayedContact} />
```

Comment:
Reading this make me feel that is a correct abstraction: `<ContactCard />` component needs a `contact` props. 👍